test/python/utils/message.py

```diff
@@ -168,6 +169,46 @@ def raw(self) -> bytes:
         raw += self.block_payload_hash
         return raw
 
+class AttestationDal(Message):
+    """Class representing an attestation + DAL."""
+    chain_id: int
+    branch: bytes
+    slot: int
+    op_level: int
+    op_round: int
+    block_payload_hash: bytes
+    dal_message: bytes
+
+    def __init__(self,
+                 chain_id: int,
+                 branch: Union[str,bytes],
+                 slot: int,
+                 op_level: int,
+                 op_round: int,
+                 block_payload_hash: Union[str,bytes],
+                 dal_message: Union[str,bytes]):
+        self.chain_id = chain_id
+        self.branch = scrub(branch)
+        assert_data_size("branch", self.branch, 32)
+        self.slot = slot
+        self.op_level = op_level
+        self.op_round = op_round
+        self.block_payload_hash = scrub(block_payload_hash)
+        assert_data_size("block_payload_hash", self.block_payload_hash, 32)
+        self.dal_message = scrub(dal_message)
+
+    def raw(self) -> bytes:
+        raw = b''
+        raw += MagicByte.TENDERBAKE_ENDORSEMENT.to_bytes(1, byteorder='big')
```

Comment:
Are you sure that `TENDERBAKE_ENDORSEMENT` is the right magic byte ?

In `tezos/src/proto_alpha/lib_protocol/operation_repr.ml`:
```ocaml
let of_watermark = function
  | Signature.Custom b ->
      if Compare.Int.(Bytes.length b > 0) then
        match Bytes.get b 0 with
        | '\x12' ->
            Option.map
              (fun chain_id -> Preattestation chain_id)
              (Chain_id.of_bytes_opt (Bytes.sub b 1 (Bytes.length b - 1)))
        | '\x13' ->
            Option.map
              (fun chain_id -> Attestation chain_id)
              (Chain_id.of_bytes_opt (Bytes.sub b 1 (Bytes.length b - 1)))
        | '\x14' ->
            Option.map
              (fun chain_id -> Dal_attestation chain_id)
              (Chain_id.of_bytes_opt (Bytes.sub b 1 (Bytes.length b - 1)))
        | _ -> None
      else None
  | _ -> None
```
I'm not sure whether this function is related to our operations, but I think it's better to be sure.

Reply:
Oh no, I was looking at the Dal_attestation case, whereas the PR is about Attestation + Dal.
You can resolve it if you agree.

Reply:
yeah attestation+DAL is seperate operation which has same magic byte and tag is 23.